File: web/js/examples/manim-gallery-moving-around.js
import {
  BLUE,
  LEFT,
  ORANGE,
  Scene,
  Square,
} from "../../noon-authoring.js";

export function movingAround() {
  const scene = new Scene();
  const square = new Square().setColor(BLUE).setFill(BLUE, 1);
  scene.add(square);

  scene.play(square.animate().shift(LEFT));
  scene.play(square.animate().setFill(ORANGE));
  scene.play(square.animate().scale(0.3));
  scene.play(square.animate().rotate(0.4));
  return scene;
}

export const galleryMovingAround = {
  MovingAround: movingAround,
};

File: crates/noon/examples/manim_gallery_moving_around.rs
use noon::prelude::*;
use noon_ir::encode_scene;

fn moving_around() -> Scene {
    let mut scene = Scene::new();
    let square = scene.add(
        Square::default()
            .color(BLUE)
            .set_fill(Some(BLUE), Some(1.0)),
    );

    scene
        .play(square.animate().shift(LEFT))
        .run_time(1.0)
        .unwrap();
    scene
        .play(square.animate().set_fill(Some(ORANGE), None))
        .run_time(1.0)
        .unwrap();
    scene
        .play(square.animate().scale(0.3))
        .run_time(1.0)
        .unwrap();
    scene
        .play(square.animate().rotate(0.4))
        .run_time(1.0)
        .unwrap();
    scene
}

fn main() {
    let scene = moving_around();
    println!(
        "MovingAround\t{}",
        encode_scene(scene.definition()).unwrap()
    );
}
